java/code/src/com/redhat/rhn/domain/image/ImageInfoFactory.java

```diff
@@ -84,44 +88,40 @@ protected Logger getLogger() {
     }
 
     /**
-     * Schedule an Image Build
+     * Create ImageInfo
      * @param buildHostId The ID of the build host
      * @param version the tag/version of the resulting image
      * @param profile the profile
-     * @param earliest earliest build
-     * @param user the current user
-     * @return the action ID
-     * @throws TaskomaticApiException if there was a Taskomatic error
-     * (typically: Taskomatic is down)
+     * @return the ImageInfo
      */
-    public static Long scheduleBuild(long buildHostId, String version, ImageProfile profile,
-            Date earliest, User user) throws TaskomaticApiException {
+    public static ImageInfo createImageInfo(long buildHostId, String version, ImageProfile profile) {
         MinionServer server = ServerFactory.lookupById(buildHostId).asMinionServer().get();
 
-        if ((!server.hasContainerBuildHostEntitlement() && profile.asDockerfileProfile().isPresent()) ||
-                (!server.hasOSImageBuildHostEntitlement() && profile.asKiwiProfile().isPresent())) {
+        boolean isDocker = profile.asDockerfileProfile().isPresent();
+        boolean isKiwi = profile.asKiwiProfile().isPresent();
+
+        if ((!server.hasContainerBuildHostEntitlement() && isDocker) ||
+                (!server.hasOSImageBuildHostEntitlement() && isKiwi)) {
             throw new IllegalArgumentException("Server is not a build host.");
         }
 
-        if (profile.asDockerfileProfile().isPresent()) {
+        if (isDocker) {
             version = version.isEmpty() ? "latest" : version;
         }
 
-        // Schedule the build
-        ImageBuildAction action = ActionManager.scheduleImageBuild(user,
-                Collections.singletonList(server.getId()), version, profile, earliest);
-        taskomaticApi.scheduleActionExecution(action);
-
-        // Load or create an image info entry
-        ImageInfo info =
-                lookupByName(profile.getLabel(), version, profile.getTargetStore().getId())
-                .orElseGet(ImageInfo::new);
+        // Create an image info entry
+        ImageInfo info = new ImageInfo();
 
-        info.setName(profile.getLabel());
-        info.setVersion(version);
+        if (isDocker) {
+            info.setName(profile.getLabel());
+            info.setVersion(version);
+        }
+        if (isKiwi) {
+            info.setName("Profile: " + profile.getLabel());
+            info.setVersion("TBD");
```

Comment:
What about setting version to null. Wouldn't that be printed as question mark icon in web ui? 'TBD' may not be immediately clear, particularly to non-english speakers with localized suma.

Reply:
null is shown as empty string, maybe it is enough